File: lib/Queue.js
class Queue {
  constructor(message) {
    this.guildID = message.guild.id;

    this.dispatcher = null;
    this.voiceConnection = null;

    this.tracks = [];

    this.stream = null;

    this.volume = 100;

    this.isPlaying = false;

    this.paused =
      this.voiceConnection &&
      this.voiceConnection.dispatcher &&
      this.voiceConnection.dispatcher.paused;

    this.firstMessage = message;
  }
  get playing() {
    return this.tracks[0];
  }
}
module.exports = Queue;

File: lib/Queue.py
class Queue:
    def __init__(self, message, connection):
        self.guild_id = message.guild.id
        self.voice_connection = connection
        self.player = None

        self.volume = 50
        self.bitrate = int(connection.channel.bitrate / 1000)
        self.is_playing = False
        self.loop = 0

        self.tracks = []
        self.filters = []

        self.filters_update = False
        self.first_message = message

        self.queue_message = None
        self.queue_page = 0

        self.now_playing = None
        self.now_playing_index = 0

        self.start_time = 0
        self.end_time = 0
